backend/app/routers/authorization.py

```diff
@@ -18,23 +23,62 @@ async def save_authorization(
     user=Depends(keycloak_auth.get_current_username),
     db: MongoClient = Depends(dependencies.get_db),
 ):
-    """Save authorization info in Mongo. This is a triple of dataset_id/user_id/role."""
-    authorization_db = AuthorizationDB(**authorization_in.dict(), creator=user)
+    """Save authorization info in Mongo. This is a triple of dataset_id/user_id/role/group_id."""
+
+    # Check all connection and abort if any one of them is not available
+    if db is None:
+        raise HTTPException(status_code=503, detail="Service not available")
+        return
+
+    # Retrieve users from groups in mongo
+    user_ids: List[EmailStr] = []
+    group_q_list = db["groups"].find({"_id": {"$in": authorization_in.group_ids}})
+    if group_q_list is not None:
+        async for group_q in group_q_list:
+            group = GroupOut.from_mongo(group_q)
+            user_ids = list(map(lambda user: user.user.email, group.userList))
+    else:
+        raise HTTPException(
+            status_code=404, detail=f"Group {authorization_in.group_ids} not found"
+        )
+
+    authorization_dict = authorization_in.dict()
+    authorization_dict["user_ids"] = list(set(user_ids))
```

Comment:
Is this section adding the users from each group to the array of users in authorization document? If so, why do we need to do this?

Reply:
I added list(set(user_ids) so that if we add same user_ids twice, I want to remove the duplicate user_id